projects/angular/src/forms/common/label.ts

```diff
@@ -45,10 +42,19 @@ export class ClrLabel implements OnInit, OnDestroy {
     private el: ElementRef<HTMLLabelElement>
   ) {}
 
+  @HostBinding('attr.id')
+  get idAttr() {
+    return this.labelId;
+  }
+
   get labelText(): string {
     return this.el.nativeElement && this.el.nativeElement.textContent;
   }
 
+  get labelId(): string {
+    return this.el.nativeElement.getAttribute('id') || this.controlIdService?.id + '-label';
```

Comment:
```suggestion
    return this.idAttr || this.controlIdService?.id + '-label';
```

We should be able to use the host binding instead of accessing the DOM directly here. We have to work out removing the custom get accessor though.